Library/Homebrew/utils/tty.rb

```diff
@@ -0,0 +1,45 @@
+module Tty
+  module_function
+
+  def strip_ansi(str)
+    str.gsub(/\033\[\d+(;\d+)*m/, "")
+  end
+
+  def width
+    `/usr/bin/tput cols`.strip.to_i
+  end
+
+  def truncate(str)
+    (w = width).zero? ? str.to_s : str.to_s[0, w - 4]
+  end
+
+  {
+    # Colors
+    red:           31,
+    green:         32,
+    yellow:        33,
+    blue:          34,
+    magenta:       35,
+    cyan:          36,
+    default:       39,
+
+    # Styles
+    reset:          0,
+    bold:           1,
+    italic:         3,
+    underline:      4,
+    strikethrough:  9,
+  }.each do |name, code|
```

Comment:
I'd consider declaring this as two frozen Hash constants and iterating over that; makes it a big more explicit what's going on.